Automattic/wp-calypso
# Hosting onboarding: Customising the "included with your purchase" list of the hosting flow

Closes Automattic/dotcom-forge#2125. Closes https://github.com/Automattic/dotcom-forge/issues/2121.

## Proposed Changes

Updates the "Included with your purchase" list in checkout for users who are going through the `/start/hosting` onboarding flow so that it includes more technical focused features. Like they would have seen on the hosting LOHP.

**Business plan**
![bus](https://user-images.githubusercontent.com/1500769/231942883-e384be2b-88eb-4102-95d9-1fe79f85518a.png)


**Commerce plan**
![comm](https://user-images.githubusercontent.com/1500769/231942903-be8a389d-046b-4b7c-9e0f-46648ba4c132.png)

Usually the items in the list are based on what's in the cart and so can be different based on whether for example there's a domain in the cart. However the ability to customise the list based on onboarding flow had already been added for the "tailored onboarding flows". We just need to update the value of `shouldUseFlowFeatureList` so it's also true for the hosting flow.

One difference between the tailored onboarding flows and the regular flows is that the flow-based feature list didn't include the refund window. But that's something we're trying to emphasise in the hosting flow. So I made sure that it's included in our flow.

One downside of basing the list on the flow and not the cart contents is that we don't highlight the fact that the user is purchasing a domain. Usually a business plan would see a list like this:
![biz-with-domain](https://user-images.githubusercontent.com/1500769/231943791-a1d06463-2a24-4f05-be79-647da468affd.png)

I don't think this is a huge deal. But worth pointing out.

## Testing Instructions



* Test the `/setup/link-in-bio` flow and ensure the checkout experience hasn't changed
* Test the regular `/start/onboarding` flow and ensure the checkout experience hasn't changed
* Test the `/start/hosting` flow
   * Test with and without domains
   * Test with business plan
   * Test with commerce plan
* Test the [etk](https://github.com/Automattic/wp-calypso/pull/75679#issuecomment-1506558301) and [happy-blocks](https://github.com/Automattic/wp-calypso/pull/75679#issuecomment-1506558898) builds
